dd-sdk-android/src/test/kotlin/com/datadog/android/rum/internal/domain/event/RumEventMapperTest.kt

```diff
@@ -456,6 +456,109 @@ internal class RumEventMapperTest {
         )
     }
 
+    @Test
+    fun `M use the original event W map returns a copy { ViewEvent }`(forge: Forge) {
+        // GIVEN
+        val fakeRumEvent = forge.getForgery<ViewEvent>()
+        whenever(mockViewEventMapper.map(fakeRumEvent))
+            .thenReturn(fakeRumEvent.copy())
+
+        // WHEN
+        val mappedRumEvent = testedRumEventMapper.map(fakeRumEvent)
+
+        // THEN
+        assertThat(mappedRumEvent).isEqualTo(fakeRumEvent)
```

Comment:
we need to use `isSameAs` here, because this line will be green even if another instance is returned (`isEqualTo` will call `equals`).

```suggestion
        assertThat(mappedRumEvent).isEqualTo(fakeRumEvent)
```

The `log` verification below helps though to avoid this test to pass-by.